cmake/ClangFormat.cmake

```diff
@@ -0,0 +1,79 @@
+function(clang_format_setup)
+  find_program(CLANG_FORMAT_EXECUTABLE NAMES clang-format)
+  if(NOT CLANG_FORMAT_EXECUTABLE)
+    message(STATUS "Looking for clang-format - NOT found, please install "
+      "clang-format to enable automatic code formatting."
+    )
+    return()
+  endif()
+
+  message(STATUS "Found clang-format.")
+endfunction()
+
+function(clang_format_setup_version clang_version)
+  find_program(CLANG_FORMAT_EXECUTABLE NAMES clang-format-${clang_version})
+  if(NOT CLANG_FORMAT_EXECUTABLE)
+    message(STATUS "Looking for clang-format-${clang_version} - NOT found,"
+      "please install to enable automatic code formatting."
+    )
+    return()
+  endif()
+
+  message(STATUS "Found clang-format-${clang_version}.")
+endfunction()
```

Comment:
7Suggestion: `clang_format_setup` and `clang_format_setup` can be merged by using [`cmake_parse_arguments`](https://cmake.org/cmake/help/v3.12/command/cmake_parse_arguments.html).